test/org/sagebionetworks/bridge/play/controllers/ParticipantControllerTest.java

```diff
@@ -516,6 +520,41 @@ public void partialUpdateSelfParticipant() throws Exception {
         assertEquals("simpleStringChange", captured.getExternalId());
     }
     
+    @Test
+    public void updateSelfParticipantReturnsParticipantAsSaved() throws Exception {
+        StudyParticipant originalParticipant = new StudyParticipant.Builder()
+                .withId(ID)
+                .withPhone(TestConstants.PHONE)
+                .withEmail(EMAIL).build();
+        doReturn(originalParticipant).when(mockParticipantService).getParticipant(eq(study), eq(ID), anyBoolean());
+        doReturn(new UserSession(originalParticipant)).when(authService).getSession(eq(study), any());
+        
+        // Simulate the case where the service does not update the participant, as can now happen. These
+        // updates are NOT returned from the service, the original participant is, and *that's* what should
+        // be used in the return from this controller, not the participant as submitted.
+        Phone updatedPhone = new Phone("408-258-8569", "US");
+        StudyParticipant updatedParticipant = new StudyParticipant.Builder()
+                .copyOf(TestUtils.getStudyParticipant(ParticipantControllerTest.class))
+                .withPhone(updatedPhone)
+                .withEmail("updated@email.com").build();
+        
+        String json = MAPPER.writeValueAsString(updatedParticipant);
+        mockPlayContextWithJson(json);
+
+        Result result = controller.updateSelfParticipant();
+        
+        assertEquals(200, result.status());
+        JsonNode node = TestUtils.getJson(result);
+        assertEquals("UserSessionInfo", node.get("type").textValue());
+        assertEquals(TestConstants.PHONE.getNumber(), node.get("phone").get("number").textValue());
+        assertEquals(EMAIL, node.get("email").textValue());
```

Comment:
Should we add a test case for when the participant is updated? Since that's the case that we're changing the code for.

Reply:
It would be the same, as the logic about whether the participant is changed or not is in the service. We want to verify that we reload the participant object rather than use the one submitted, so that's why the test is verifying if the participant object is different from what was submitted, that's what we see cached and returned.